system/metrics/sinks/common/periodicSink.cpp

```diff
@@ -0,0 +1,83 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
```

Comment:
This makes sense to be commoned up - but should probably live in jmetrics.cpp along with the MetricSink base class.  (It would avoid including it in multiple projects.)